src/Symfony/Component/Security/Http/Firewall/AbstractPreAuthenticatedListener.php

```diff
@@ -54,6 +54,15 @@ public function __construct(TokenStorageInterface $tokenStorage, AuthenticationM
      */
     final public function handle(GetResponseEvent $event)
     {
+        $token = $this->tokenStorage->getToken();
+        if(null !== $token && false === ($token instanceof PreAuthenticatedToken) && $token->isAuthenticated()) {
```

Comment:
I personnally don't like the syntax
` !$token instanceof PreAuthenticatedToken`
It's correct thanks to operator precedence, but not obvious, but I checked in the current code and it's the more used.

I couldn't find any mention about it in the Symfony coding standard, did I miss it?
It would be nice to add.